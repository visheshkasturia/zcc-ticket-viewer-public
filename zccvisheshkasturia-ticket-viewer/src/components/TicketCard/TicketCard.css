.ticketContainer{
  display: grid;
  grid-template-columns: 10% 10% 10% 50% 20%;
  grid-template-rows: 100;
  border: 1px solid black;
  min-height: 70px; 
  width: 700px;
  margin-left: auto;
  margin-right: auto;
  margin-bottom: 10px;
}

.flex-center{
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 0 5px;
}

.id-box{
  grid-column: 1/ span 1;
  border: 1px solid black;
}
.status-box{
  grid-column: 2/ span 1;
  border: 1px solid black;
}
.priority-box{
  grid-column: 3/ span 1;
  border: 1px solid black;
}
.subject-box{
  grid-column: 4/ span 1;
  border: 1px solid black;
}
.date-box{
  grid-column: 5/ span 1;
  border: 1px solid black;
}