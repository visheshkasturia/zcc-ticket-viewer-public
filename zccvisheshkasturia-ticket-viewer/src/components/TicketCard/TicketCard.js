import React, { useState } from 'react';
import TicketDetails from '../TicketDetails/TicketDetails';
import './TicketCard.css';
import PopUp from '../PopUp/PopUp';

function TicketCard({ ticket }) {
  const [showDetails, setShowDetails] = useState(false);
  const handleShowDetails = (e) => {
    e.preventDefault();
    setShowDetails(!showDetails);
  };

  return (
    <div className="ticketContainer" onClick={handleShowDetails} onKeyDown={() => { setShowDetails(!showDetails); }} role="button" tabIndex={0}>
      <div data-testid="ticket-id-test" className="id-box flex-center">{ticket.id}</div>
      <div data-testid="ticket-status-test" className="status-box flex-center">{ticket.status}</div>
      <div data-testid="ticket-priority-test" className="priority-box flex-center">{ticket.priority}</div>
      <div data-testid="ticket-subject-test" className="subject-box flex-center">{ticket.subject}</div>
      <div data-testid="ticket-createdat-test" className="date-box flex-center">{ticket.created_at.slice(0, ticket.created_at.indexOf('T'))}</div>
      {showDetails
      && <PopUp content={<TicketDetails ticket={ticket} />} handleClose={handleShowDetails} />}
    </div>
  );
}

export default TicketCard;
