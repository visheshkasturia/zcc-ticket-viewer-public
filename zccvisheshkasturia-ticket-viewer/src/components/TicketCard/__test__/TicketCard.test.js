import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import TicketCard from '../TicketCard';

afterEach(cleanup);
describe('TicketCard renders correctly', () => {
  const ticketData = {
    id: 1,
    status: 'open',
    priority: 'urgent',
    subject: 'test subject',
    created_at: '2021-12-01T17:53:53Z',
  };

  it('id is correct', () => {
    const { getByTestId } = render(<TicketCard ticket={ticketData} />);
    expect(getByTestId('ticket-id-test')).toHaveTextContent('1');
  });

  it('status is correct', () => {
    const { getByTestId } = render(<TicketCard ticket={ticketData} />);
    expect(getByTestId('ticket-status-test')).toHaveTextContent('open');
  });

  it('priority is correct', () => {
    const { getByTestId } = render(<TicketCard ticket={ticketData} />);
    expect(getByTestId('ticket-priority-test')).toHaveTextContent('urgent');
  });

  it('subject is correct', () => {
    const { getByTestId } = render(<TicketCard ticket={ticketData} />);
    expect(getByTestId('ticket-subject-test')).toHaveTextContent('test subject');
  });

  it('date is correct', () => {
    const { getByTestId } = render(<TicketCard ticket={ticketData} />);
    expect(getByTestId('ticket-createdat-test')).toHaveTextContent('2021-12-01');
  });

  it('Matches snapshot', () => {
    const tree = renderer.create(<TicketCard ticket={ticketData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
