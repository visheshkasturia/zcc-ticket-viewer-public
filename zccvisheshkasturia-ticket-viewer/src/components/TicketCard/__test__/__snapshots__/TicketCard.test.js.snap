// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TicketCard renders correctly Matches snapshot 1`] = `
<div
  className="ticketContainer"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="id-box flex-center"
    data-testid="ticket-id-test"
  >
    1
  </div>
  <div
    className="status-box flex-center"
    data-testid="ticket-status-test"
  >
    open
  </div>
  <div
    className="priority-box flex-center"
    data-testid="ticket-priority-test"
  >
    urgent
  </div>
  <div
    className="subject-box flex-center"
    data-testid="ticket-subject-test"
  >
    test subject
  </div>
  <div
    className="date-box flex-center"
    data-testid="ticket-createdat-test"
  >
    2021-12-01
  </div>
</div>
`;
